blockchain/ethereum/ledger/ledger.go

```diff
@@ -0,0 +1,342 @@
+package ledger
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/republicprotocol/republic-go/blockchain/ethereum"
+	"github.com/republicprotocol/republic-go/blockchain/ethereum/bindings"
+	"github.com/republicprotocol/republic-go/order"
+)
+
+const BlocksForConfirmation = 1
+
+// RenLedgerContract
+type RenLedgerContract struct {
+	network      ethereum.Network
+	context      context.Context
+	conn         ethereum.Conn
+	transactOpts *bind.TransactOpts
+	callOpts     *bind.CallOpts
+	binding      *bindings.RenLedger
+	address      common.Address
+}
+
+// NewRenLedgerContract creates a new NewRenLedgerContract with given parameters.
+func NewRenLedgerContract(ctx context.Context, conn ethereum.Conn, transactOpts *bind.TransactOpts, callOpts *bind.CallOpts) (RenLedgerContract, error) {
+	contract, err := bindings.NewRenLedger(common.HexToAddress(conn.Config.RenLedgerAddress), bind.ContractBackend(conn.Client))
+	if err != nil {
+		return RenLedgerContract{}, err
+	}
+
+	return RenLedgerContract{
+		network:      conn.Config.Network,
+		context:      ctx,
+		conn:         conn,
+		transactOpts: transactOpts,
+		callOpts:     callOpts,
+		binding:      contract,
+		address:      common.HexToAddress(conn.Config.RenLedgerAddress),
+	}, nil
+}
+
+func (ledger *RenLedgerContract) OpenBuyOrder(signature [65]byte, id order.ID) error {
+
+	ledger.transactOpts.GasPrice = big.NewInt(int64(5000000000))
```

Comment:
Should we allow them to set the gas price? Not sure if 5 gwei is good for all case.

Reply:
I think having a default is a good idea, but 5GWEI is probably too low. Review task has been added.